Add project documentation files: README.md, board_rules.md, CHANGELOG.md, release_notes.md. (#25)
ios-course/ironfoudation-team-project: board_rules.md

# Board rules 

The description of the team workflow for the board.

## Description

The board consists of 5 columns:

- `Common backlog`- includes all tasks that should be done.
- `Sprint backlog` - includes the list of tasks for the current sprint.
- `In Progress` - the list of tasks that the team works on.
- `In Review` - the list of tasks that are under review process.
- `Done` - the list of tasks that are considered as completed. If a task is no longer makes sense, then it must be closed with an appropriate comment after agreement with the team.

### Usage

The issue moves from left to right and can't skip any step (board section). The exception is when the team decides that the task cannot be completed or it needs to be canceled (read the description of **Done** section).
The main flow is:
 `
Common backlog -> Sprint backlog -> In progress -> In review -> Done.

> `

Reviewer: Seems a typo.

```suggestion
```

Author: Fixed it.

### Common backlog

Every issue is placed in a common backlog when it is created.
The issue must contain a clear title and description with definition of done (**DoD**).
Select necessary labels, if applicable. For example, if you add a task to create documentation, put the label "Documentation" ([label's description](https://github.com/ios-course/ironfoudation-team-project/labels)). 
Don't forget to choose corresponding project in the `Projects` section on pages, describing PR's and issue's contain.

### Sprint backlog

Before sprint planning the issues place into Sprint backlog by PO.

> A milestone should be put to the issue at the moment of moving it into the Sprint backlog.

Reviewer: Can I do it earlier?
I'd suggest:
```suggestion
Issues in the Sprint backlog must have an assigned milestone.
```

Author: Thanks for the wording. Sounds much better!)
Fixed it.

During the sprint planning process, the issues from the sprint backlog are estimated (labels with estimation are assigned).

**Estimation** is carried out with the participation of all team members online using the **Scrum Poker** technology on the resource [planningpokeronline](https://planningpokeronline.com).

> Except estimation labels in project also provided [labels](https://github.com/ios-course/ironfoudation-team-project/labels). 

Reviewer: nit: we had the same link above already. Do we need this paragraph?

Author: Yes, you are right. This is duplicate information. 
The paragraph is removed.

**Important**: 

* the fields milestone, and labels with estimation points should be filled for the issues after Sprint planning.

### In Progress

When a team member start working on the task, they must move it into the `In Progress` section. Issue stays there until PR will be opened. If issue doesn't require PR to be completed after agreement with the team its successful completion it will be moved into the `Done` section.

> **Important**: the field assignee should be immediately filled when the task'll be moved `In Progress` section.

Reviewer: nit:
```suggestion
**Important**: Issues in the `In Progress` section must have at least one assignee who is responsible for the issue.
```

Author: Thanks! Fixed it.

### In Review

When the task code is under review, it is located in `In Progress` column.
When the work on the task is done, the PR is opened and waited for the review process, the PR is placed in the in Review section. There are only open PRs in the `In Review` column.

> Don't forget to link the opened PR with the issue. For doing that, please, fill the `Development section' when describing the issue and choose menu item "Link a branch or pull request" -> select the a repository and PR.

Reviewer: ```suggestion
Don't forget to link the opened PR with the issue. For doing that, please, fill the `Development section` when describing the issue and choose menu item "Link a branch or pull request" -> select the a repository and PR.
```

Author: Thanks for the attention!) 
Fixed it.

# Board rules 

 The description of the team workflow for the board.

## Description

The board consists of 5 columns:
```
**Common backlog** - includes all tasks that should be done.
**Sprint backlog** - includes the list of tasks for the current sprint.
**In Progress** - the list of tasks that the team works on.
**In Review** - the list of tasks that are under review process.
**Done** - the list of tasks that successfully merged to main branch.

> ```

Author: Thanks, fixed it.

### Common backlog

The process of the issue creating and adding to Common backlog.
Every issue is placed in a common backlog when it is created.
The issue should contain a clear title and description with definitions of done (**DoD**).
Select necessary labels, if applicable.
Don't forget to link the project Repository.

### Sprint backlog

Before sprint planning the issues place into Sprint backlog by PM.
During the sprint planning process, the issues from the sprint backlog are estimated (labels with estimation are assigned), and the assignee is defined. 

> **Estimation** is carried out with the participation of all team members online using the **Scrum Poker** technology on the resource:
> ```
>  https://planningpokeronline.com
>  ``` 

Reviewer 2: How about using the link format?

```suggestion
**Estimation** is carried out with the participation of all team members online using the **Scrum Poker** technology on the resource [planningpokeronline](https://planningpokeronline.com)

```

Author: Thanks, fixed it.

Author: Thanks, fixed it.

> Except estimation labels in progect also provided labels 
> ```high, medium, bug, blocker, enhancement, help wanted.
> ```
> All types of labels are marked with separate color.

Reviewer 2: It doesn't display properly. Please check the markdown format.
Maybe we need to add a description for every label, that we used.
```suggestion
Except estimation labels in progect also provided labels: 

`high`, `medium`, `bug`, `blocker`, `enhancement`, `help wanted`.

All types of labels are marked with separate color.
```

Author: Thanks, the description for labels is added.

> **Important**: the fields assignees, milestone, and labels with estimation points should be filled for the issues after sprint planning.

Author: Thanks, the information updated.

### In Progress

When the work on the task starts it moves to in Progress section and stays there until PR will be opened.

### In Review

When the work on the task is done, the PR is opened and waited for the review process, the issue is placed in the in Review section.
Don't forget to link the opened PR with the issue.

### Done

The issues have successfully gone through the review process and merged into main branch can be moved to the Done section.

**It also important to know:**
To process application,some private cliets information like home adress, mobile phone is needed.

> 

Reviewer 2: Do we need this here?

## Usage

The issue moves from left to right and can't skip any step (board section).
The main flow is:
 ```
Common backlog -> Sprint backlog -> In progress -> In review -> Done.
```

## Ironfoundation-team

You can contact the authors by the following email adresses:
```
Nadia Shpakouskaya: shpakovna@gmail.com 
Alex Zavadskiy: fh.zavadskii@gmail.com
Julya Kmita: julyakmita@gmail.com

> ```

Author: The emails are removed. Sorry for this!

### In Progress

When the work on the task starts it moves to in Progress section and stays there until PR will be opened.

**Important**: the field assignee should be strictly added `In Progress` section. 

### In Review

When the task code is under review, it is located in Progress column.
When the work on the task is done, the PR is opened and waited for the review process, the PR is placed in the in Review section. There are only open PRs in the `In Review` column.
Don't forget to link the opened PR with the issue. For doing that, please, fill the "Development section" when describing the issue and choose menu item "Link a branch or pull request" -> select the a repository and PR.
**Important**: if the PR is approved, then the PR creator can independently resolve conversations. 

### Done

`Done` - the list of tasks that successfully finished. If a task contains code, it should be merged to the main branch. In the case, when a task is useless or can not be closed for some resons it should be closed in the next mode "Close as not planned" (won't fix, can't repro, duplicate, stale etc.).

> 

Author: Thanks!) Great improvement! Fixed it.

# Board rules 

The description of the team workflow for the board.

## Description

The board consists of 5 columns:

- `Common backlog`- includes all tasks that should be done.
- `Sprint backlog` - includes the list of tasks for the current sprint.
- `In Progress` - the list of tasks that the team works on.
- `In Review` - the list of tasks that are under review process.

> - `Done` - the list of tasks that were successfully finished. If a task contains code, it should be merged into the main branch. In the case, when a task is useless or can not be closed for some reason it should be closed in the next mode "Close as not planned" (won't fix, can't repro, duplicate, stale, etc.).

Author: Thanks, fixed it.

### Common backlog

> The process of the issue creating and adding to Common backlog.

Author: The line is deleted.

Every issue is placed in a common backlog when it is created.

> The issue should contain a clear title and description with definitions of done (**DoD**).

Author: Thanks, fixed it.

> The issue should contain a clear title and description with definitions of done (**DoD**).

Author: Thanks, fixed it.

Select necessary labels, if applicable. For example, if you add a task to create documentation, put the label "Documentation" (label's description is provided below). 

> Don't forget to link the project Repository.

Author: Fixed the description.

### Sprint backlog

> Before sprint planning the issues place into Sprint backlog by PM.

Author: Yes, we should. Fixed it.

During the sprint planning process, the issues from the sprint backlog are estimated (labels with estimation are assigned).

**Estimation** is carried out with the participation of all team members online using the **Scrum Poker** technology on the resource [planningpokeronline](https://planningpokeronline.com).

Except estimation labels in project also provided labels: 

> - `Invalid`- this doesn't seem right.

Author: Yes, you are right. Thanks, fixed it.

- `Documentation` - improvements or additions to documentation.
- `Need more discussion`- further information is requested.
- `Bug`- something isn't working.
- `Blocker` - urgent level of priority, it's blocking another task.
- `Blocked` - this task cannot be done until the task on which it depends is completed. (Inside the blocked issue is placed a link to the blocker).

> All types of labels are marked with separate color.

Author: The line is removed.

**Important**: 

> * the fields milestone, and labels with estimation points should be filled for the issues after Sprint planning.

Author: My idea was to highlight important information additional information in each section if any.   This information belongs to the last paragraph and would be unreadable elsewhere, in my opinion. If you don't mind, I would leave it here. Or should I move it?

### In Progress

> When the work on the task starts it moves to in Progress section and stays there until PR will be opened.

Reviewer: ```suggestion
When a team member start working on the task, they must move it into the `In Progress` section. Issue stays there until PR will be opened.
```

Author: Thanks, fixed it.

> When the work on the task starts it moves to in Progress section and stays there until PR will be opened.

Reviewer: What if issue doesn't require PR to be completed?

Author: Fixed it.

> **Important**: the field assignee should be strictly added `In Progress` section. 

Reviewer 3: @julyakmita what you meant?
```suggestion
**Important**: the field assignee should be immediately filled when the task'll be moved `In Progress` section.
```

### In Review

> When the task code is under review, it is located in Progress column.

Reviewer: we don't have "Progress" column.

Author: Sorry for the typo. Of course, we have the column `In Progress`.
Fixed it.

When the work on the task is done, the PR is opened and waited for the review process, the PR is placed in the in Review section. There are only open PRs in the `In Review` column.
Don't forget to link the opened PR with the issue. For doing that, please, fill the "Development section" when describing the issue and choose menu item "Link a branch or pull request" -> select the a repository and PR.
**Important**: if the PR is approved, then the PR creator can independently resolve conversations. 

### Done

> An issue containing code can be moved to the `Done` section after a successful merge process with the main branch.

Reviewer: > An issue containing code can be moved...

How the issue can contain code?

Author: Fixed the description.

> An issue containing code can be moved to the `Done` section after a successful merge process with the main branch.

Author: Thanks, fixed it.

If the issue does not contain code and does not require a merge process, it can be closed by skipping the `In Review` section and moving directly from the `In Progress` section to the `Done` section.

> In case the issue is useless, duplicates some other issues, or is not planned to be fixed, it should be closed in the next mode "Close as not scheduled" (does not fix, cannot reproduce, duplicates, outdated, etc.). Additional comments about why the issue is closed are required.

Reviewer: We have this description above. Why do we duplicating it?

Author: Agree, a duplicate lines are deleted.

> ## Usage

Reviewer: nit:
Should it be placed before the description of the columns?

The issue moves from left to right and can't skip any step (board section). The exception is when the team decides that the task cannot be completed or it needs to be canceled (read the description of **Done** section).
The main flow is:
 `
Common backlog -> Sprint backlog -> In progress -> In review -> Done.
`

## Important:

> ### Creating a new branch:

Reviewer: Why we have the section `Creating a new branch` in the `board_rules.md` file?

Author: You are right. This information doesn't apply to this file. It is removed.

# Board rules 

The description of the team workflow for the board.

## Description

The board consists of 5 columns:

- `Common backlog`- includes all tasks that should be done.
- `Sprint backlog` - includes the list of tasks for the current sprint.
- `In Progress` - the list of tasks that the team works on.
- `In Review` - the list of tasks that are under review process.

> - `Done` - the list of tasks that successfully finished. If a task contains code, it should be merged to the main branch. In the case, when a task is useless or can not be closed for some resons it should be closed in the next mode "Close as not planned" (won't fix, can't repro, duplicate, stale etc.).
> 

Reviewer 2: fix typos
```suggestion
- `Done` - the list of tasks that were successfully finished. If a task contains code, it should be merged into the main branch. In the case, when a task is useless or can not be closed for some reason it should be closed in the next mode "Close as not planned" (won't fix, can't repro, duplicate, stale, etc.).

```

### Common backlog

The process of the issue creating and adding to Common backlog.
Every issue is placed in a common backlog when it is created.
The issue should contain a clear title and description with definitions of done (**DoD**).

> Select necessary labels, if applicable. For example, if you add a task to create documentation, put the lable "Documentation" (labels' descriptiopn is provided below). 

Reviewer 3: typo:
```suggestion
Select necessary labels, if applicable. For example, if you add a task to create documentation, put the label "Documentation" (label's description is provided below). 
```

Author: Sorry for this! Thanks, fixed it.

Don't forget to link the project Repository.

### Sprint backlog

Before sprint planning the issues place into Sprint backlog by PM.
During the sprint planning process, the issues from the sprint backlog are estimated (labels with estimation are assigned).

**Estimation** is carried out with the participation of all team members online using the **Scrum Poker** technology on the resource [planningpokeronline](https://planningpokeronline.com).

> Except estimation labels in progect also provided labels: 

Reviewer 3: typo:
```suggestion
Except estimation labels in project also provided labels: 
```

Author: Sorry for this! Thanks for your attention! Fixed it.

- `Invalid`- this doesn't seem right.
- `Documentation` - improvements or additions to documentation.
- `Need more discussion`- further information is requested.
- `Bug`- something isn't working.
- `Blocker` - urgent level of priority, it's blocking another task.
- `Blocked` - this task cannot be done until the task on which it depends is completed. (Inside the blocked issue is placed a link to the blocker).

All types of labels are marked with separate color.

**Important**: 

* the fields milestone, and labels with estimation points should be filled for the issues after Sprint planning;

> * if the PR is approved, then the PR creator can independently resolove conversations. 

Reviewer 2: Maybe we should move this line to `In review` section.

Author: Thanks, fixed it.

> * if the PR is approved, then the PR creator can independently resolove conversations. 

Reviewer 3: typo:
```suggestion
* if the PR is approved, then the PR creator can independently resolve conversations. 
```

### In Progress

When the work on the task starts it moves to in Progress section and stays there until PR will be opened.

**Important**: the field assignee should be strictly added `In Progress` section. 

### In Review

When the task code is under review, it is located in Progress column.

> When the work on the task is done, the PR is opened and waited for the review process, the PR is placed in the in Review section. In the review column there are only open PR.

Reviewer 2: nit:
```suggestion
When the work on the task is done, the PR is opened and waited for the review process, the PR is placed in the in Review section. There are only open PRs in the `In Review` column.
```

> Don't forget to link the opened PR with the issue. For doing that, please, fill the "Develoment section" when describing the issue and choose menu item "Link a branch or pull request" -> select the a repository and PR.

Reviewer 3: typo:
```suggestion
Don't forget to link the opened PR with the issue. For doing that, please, fill the "Development section" when describing the issue and choose menu item "Link a branch or pull request" -> select the a repository and PR.
```

Author: Sorry for this! Thanks, fixed it.

### Done

> The issues have successfully gone through the review process and merged into main branch can be moved to the Done section.

Reviewer 2: Please update the info here as well, as in Description section.
- `Done` - the list of tasks that successfully finished. If a task contains code, it should be merged to the main branch. In the case, when a task is useless or can not be closed for some resons it should be closed in the next mode "Close as not planned" (won't fix, can't repro, duplicate, stale etc.).

Author: Thanks. fixed it.

## Usage

The issue moves from left to right and can't skip any step (board section). The exception is when the team decides that the task cannot be completed or it needs to be canceled (read the description of **Done** section).
The main flow is:
 `
Common backlog -> Sprint backlog -> In progress -> In review -> Done.
`

## Important:

### Creating a new branch:
1) all branches should be created from the main branch;
2) name of a new branch should show the issue type(feature, bug fix), issue number, and branch name and follow this pattern  [type_of_task - #issue_from_backlog - title]. 
For example: **feature-1-create-project.**

### Merge restrictions to the main branch:
- minimum two approvals;
- all conversations should be closed (resolved) after exhaustive discussion.

## Note

> The team has daily stand ups at ten o'clock (GTM+3).

Reviewer 2: Maybe just provide the link to the wiki page with the team schedule, to have one source of truth, and do not update the info in different places.
Or just remove this section, it's not related to board rules.

Author: Yes, you are right! This information isn't related to board rules. The line is deleted.

> The team has daily stand ups at ten o'clock (GTM+3).

Reviewer 3: typo:
```suggestion
The team has daily stand ups at ten o'clock (GMT+3).
```

Author: Thanks for your attention!) I fixed it. But would you mind if I remove this section as Nadia suggested, because it's not related to board rules?

> # Board rules 

Reviewer: Why some file names are capitalized and some are not?

Author: Because as I know, the README.md is usually written with a capital letter. For the names of other files I did not meet restrictions. Shall I renamed other files or it's optional?

Reviewer: this is optional in that case, but always be consistent: if you use CAPITALIZED names for md files, you can use such approach for all the files.

Author: Thanks, I'll keep it mind for the future.

The description of the team workflow for the board.

## Description

The board consists of 5 columns:

- `Common backlog`- includes all tasks that should be done.
- `Sprint backlog` - includes the list of tasks for the current sprint.
- `In Progress` - the list of tasks that the team works on.
- `In Review` - the list of tasks that are under review process.

> - `Done` - the list of tasks that successfully merged to main branch.

Reviewer: What if we have done the task that does not require writing code? Will it be "merged to main branch"?

What if we have realized that some task is useless and close them. Will it be "merged to main branch"?

Author: Thanks, fixed it.

### Common backlog

The process of the issue creating and adding to Common backlog.
Every issue is placed in a common backlog when it is created.
The issue should contain a clear title and description with definitions of done (**DoD**).

> Select necessary labels, if applicable.

Reviewer: What labels at this stage are necessary?
What labels are necessary at the other stages?

Author: The description is added.

Don't forget to link the project Repository.

### Sprint backlog

Before sprint planning the issues place into Sprint backlog by PM.
During the sprint planning process, the issues from the sprint backlog are estimated (labels with estimation are assigned).

**Estimation** is carried out with the participation of all team members online using the **Scrum Poker** technology on the resource [planningpokeronline](https://planningpokeronline.com).

Except estimation labels in progect also provided labels: 
`high`, `medium`, `bug`, `blocker`, `enhancement`, `help wanted`:

> - `High` - high level of priority.

Reviewer: Could you please explain when I should use the `medium` priority and when the `high`?

Which tasks are medium priority in the first sprint?

- `Invalid`- this doesn't seem right.
- `Documentation` - improvements or additions to documentation.
- `Medium` - medium level of priority.

> - `Enhancement`- new feature or request.

Reviewer: Why the `enhancement` is the new feature?
Why the bug fix can't be an `enhancement` ?

Author: We have discussed this question with the team today and decided, that we don't need the bags enhancement. It is deleted.

- `Need more discussion`- further information is requested.
- `Bug`- something isn't working.

> - `Help wanted` - extra attention is needed.

Reviewer: What does that mean "extra attention"?
When I should use the `Need more discussion` label and when the `Help wanted`?

Author: We decided to leave only one of these very similar badges: need more discussion. The badge "Help wanted" is deleted.

- `Low` - low level of priority.

> - `Blocker` - urgent level of priority, it's blocking another task.

Author: After discussion with team, we decided to have the badge
'Blocked' - this task cannot be done until the task on which it depends is completed.
 Inside the blocked issue we are going to put a link to the blocker.

All types of labels are marked with separate color.

**Important**: the fields milestone, and labels with estimation points should be filled for the issues after Sprint planning.

### In Progress

When the work on the task starts it moves to in Progress section and stays there until PR will be opened.

**Important**: the field assignee should be strictly added `In Progress` section. 

### In Review

> When the work on the task is done, the PR is opened and waited for the review process, the issue is placed in the in Review section.

Author: The description are fixed.

> Don't forget to link the opened PR with the issue.

Author: The description are fixed.

### Done

The issues have successfully gone through the review process and merged into main branch can be moved to the Done section.

## Usage

> The issue moves from left to right and can't skip any step (board section).

Reviewer: What if the task A is useless and we decide to cancel it?
Can I move it from the backlog directly to the `Done` column?

Author: Thanks, the description is fixed.

The main flow is:
 `
Common backlog -> Sprint backlog -> In progress -> In review -> Done.
`

## Ironfoundation-team

You can contact the authors by the following email adresses:

> - Nadia Shpakouskaya: [@NadzeyaShpakouskaya](https://github.com/NadzeyaShpakouskaya)

Author: The information are deleted.

- Alex Zavadskiy: [@iBYarchitect](https://github.com/iBYarchitect)
- Julya Kmita: [@julyakmita](https://github.com/julyakmita)

## Important:

### Creating a new branch:
1) all branches should be created from the main branch;

> 2) name of a new branch should show the issue type, issue number, and branch name and follow this pattern  [type_of_task - #issue_from_backlog - title]. 

Author: We have following issue types : feature, bug, bug fix.
Fixed the description.

# Board rules 

The description of the team workflow for the board.

## Description

The board consists of 5 columns:

- `Common backlog`- includes all tasks that should be done.
- `Sprint backlog` - includes the list of tasks for the current sprint.
- `In Progress` - the list of tasks that the team works on.
- `In Review` - the list of tasks that are under review process.
- `Done` - the list of tasks that successfully merged to main branch.

### Common backlog

The process of the issue creating and adding to Common backlog.
Every issue is placed in a common backlog when it is created.
The issue should contain a clear title and description with definitions of done (**DoD**).
Select necessary labels, if applicable.
Don't forget to link the project Repository.

### Sprint backlog

Before sprint planning the issues place into Sprint backlog by PM.

> During the sprint planning process, the issues from the sprint backlog are estimated (labels with estimation are assigned), and the assignee is defined.

Author: Fixed it.

**Estimation** is carried out with the participation of all team members online using the **Scrum Poker** technology on the resource [planningpokeronline](https://planningpokeronline.com).

Except estimation labels in progect also provided labels: 
`high`, `medium`, `bug`, `blocker`, `enhancement`, `help wanted`:

- `High` - high level of priority.
- `Invalid`- this doesn't seem right.
- `Documentation` - improvements or additions to documentation.
- `Medium` - medium level of priority.
- `Enhancement`- new feature or request.
- `Need more discussion`- further information is requested.
- `Bug`- something isn't working.
- `Help wanted` - extra attention is needed.
- `Low` - low level of priority.
- `Blocker` - urgent level of priority, it's blocking another task.

All types of labels are marked with separate color.

> **Important**: the field assignee should be strictly added In Progress section. The fields milestone, and labels with estimation points should be filled for the issues after Sprint planning.

Reviewer 2: Maybe we can move this info to `In Progress` section.
`**Important**: the field assignee should be strictly added In Progress section.`

Author: Yes, thanks. Removed mentioned lines.

### In Progress

When the work on the task starts it moves to in Progress section and stays there until PR will be opened.

### In Review

When the work on the task is done, the PR is opened and waited for the review process, the issue is placed in the in Review section.
Don't forget to link the opened PR with the issue.

### Done

The issues have successfully gone through the review process and merged into main branch can be moved to the Done section.

## Usage

The issue moves from left to right and can't skip any step (board section).
The main flow is:
 `
Common backlog -> Sprint backlog -> In progress -> In review -> Done.
`

## Ironfoundation-team

You can contact the authors by the following email adresses:

```
Nadia Shpakouskaya: @NadzeyaShpakouskaya 
Alex Zavadskiy: @iBYarchitect
Julya Kmita: @julyakmita

> ```

Reviewer 2: Please format this code as mentioned in `README` file as well.

Author: Fixed it.

## Important:

### Creating a new branch:
1) all branches should be created from the main branch;

> 2) name of a new branch should show the issue type, issue number and branch name:
>  
>  ```
>  [type_of_task - #issue_from_backlog - title].
>  For example:
>  feature-1-create-project.
>  ```
> 

Reviewer 2: What do you think about this formatting?
```suggestion
2) name of a new branch should show the issue type, issue number, and branch name and follow this pattern  [type_of_task - #issue_from_backlog - title]. 
For example: **feature-1-create-project.**
```

Author: It's more readable! Thanks. Agree. Fixed.

### Merge restrictions to the main branch:
- minimum two approvals;

> - all conversations should be closed (resolve) after exhaustive discussion.

Reviewer 2: Tiny typo.
```suggestion
- all conversations should be closed (resolved) after exhaustive discussion.
```

Author: Sorry for this). Thanks, fixed it.
